### PR TITLE
enhance: support generate blocks in Terragrunt

### PR DESCRIPTION
**Pull request by aliscott**
* Updates Terragrunt version
* Adds support for `generate` blocks

### CODE REVIEW

File: internal/providers/terraform/terragrunt_hcl_provider.go [hunk 1]
```diff
@@ -575,6 +574,13 @@ func (p *TerragruntHCLProvider) runTerragrunt(opts *tgoptions.TerragruntOptions)
 		}
 	}
 
+	for _, config := range terragruntConfig.GenerateConfigs {
```
Comment (aliscott) [inline internal/providers/terraform/terragrunt_hcl_provider.go:577]: @hugorut do we need to wrap this in the terragruntSourceLock?

File: go.mod [hunk 2]
```diff
@@ -252,22 +250,22 @@ require (
...
+	google.golang.org/grpc v1.55.0 // indirect
+	google.golang.org/protobuf v1.31.0 // indirect
 )
 
 replace github.com/jedib0t/go-pretty/v6 => github.com/aliscott/go-pretty/v6 v6.1.1-0.20210226104003-408905a61c8e
 
 replace github.com/spf13/cobra => github.com/spf13/cobra v1.4.0
 
-replace github.com/gruntwork-io/terragrunt => github.com/infracost/terragrunt v0.47.1-0.20230808085620-a0a234c63878
+replace github.com/gruntwork-io/terragrunt => github.com/infracost/terragrunt v0.47.1-0.20231010113615-a01919cb9fce
```
Comment (aliscott) [inline go.mod:271]: I'll update this once https://github.com/infracost/terragrunt/pull/8 is merged

File: internal/providers/terraform/terragrunt_hcl_provider.go (hunk 1 above)
Comment (hugorut) [reply internal/providers/terraform/terragrunt_hcl_provider.go:577]: yes, because it manipulates the `workingDir`

**Review by hugorut — APPROVED** [review]
looks good - let's add some locking

**Review by hugorut — APPROVED** [review]

**Pull request merged**